### DISCUSSION TITLE
Architecture diagrams

### DISCUSSION THREAD

**Discussion by IsmaelMartinez** [juspay/hyperswitch#1643] [Q&A] [answered]
Thanks for making this project open source. 

Can you provide documentation with architecture diagrams showing how everything ties together? 

Thanks in advance.

**Comment by SanchithHegde** [comment]
Hey @IsmaelMartinez, we haven't updated our architecture diagrams since going open source. We'll update it and publish it sometime soon.

**Reply by anupj** [reply]
Gentle nudge @SanchithHegde ; I think it will be very useful to understand how different crates/components work together.

**Comment by SanchithHegde** [comment, marked as answer]
Closing this as completed in #1825. You can find the architecture docs here:

https://github.com/juspay/hyperswitch/blob/main/docs/architecture.md

Let us know if anything is still unclear.

CC: @anupj
